skyportal/tests/rate_limiting/test_rate_limiting.py

```diff
@@ -0,0 +1,58 @@
+import time
+import socket
+import requests
+from baselayer.app.env import load_env
+from skyportal.tests import api
+
+
+_, cfg = load_env()
+
+s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+s.connect(("8.8.8.8", 53))
+localhost_external_ip = s.getsockname()[0]
+s.close()
+
+
+def test_api_rate_limiting(view_only_token):
+    # In case this test gets run after those defined below
+    time.sleep(1)
+    n_successful_requests = 0
+    status = 200
+    while status == 200 and n_successful_requests < 100:
```

Comment:
This is fine, but it's a bit easier to write:

```
for n_successful_requests in range(1, 100):
    r = ...
    if r.status_code == 200:
        break
```

Reply:
:+1: done